fix: Add option to update OpenAdapt app

mobley-trent opened the pull request:
<!-- Thank you for submitting a pull request! To ensure a prompt review of your changes, please provide the following information. -->

**What kind of change does this PR introduce?**

A fix that enables the OpenAdapt app to update on launch

<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Summary**

Fixes #755
/claim #755

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? Try to link to an open issue. -->

**Checklist**
* [ ] My code follows the style guidelines of OpenAdapt
* [ ] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [ ] I have linted my code locally prior to submission
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [ ] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

<!-- See the README.md for examples. Include test output. -->


**Other information**
<!-- Delete this subheading if no additional context is needed. -->

abrichr commented on openadapt/start.py:
```diff
@@ -7,7 +7,7 @@
+from openadapt.app.main import start
 
 
 def main() -> None:
```
Thank you for asking @mobley-trent ! I'm not sure. @KIRA009 @0dm thoughts? 🙏

abrichr replied:
In general we are eventually going to deprecate any file that imports nicegui.

mobley-trent commented:
@abrichr I implemented two shell scripts to handle the downloads in both windows and macos. I added the option to run the script in `openadapt/app/main.py` under `ui.icon("update")`.

I had some trouble building the application since I am running on Linux. I hope you could help me test if the feature is working.

abrichr commented:
Thank you for your contribution @mobley-trent !

Typically how applications handle this is to say something like:

> A new version is available: vxx.xx.xx (you are on vyy.yy.yy). Would you like to update? Yes/No/Skip this version

We can show a message on startup indicating that a new version is available, and let the user decide how to proceed.

If you're on Linux it will be tricky to test this out. I suggest either implementing Linux support:

https://github.com/OpenAdaptAI/OpenAdapt/issues/144

Or using something like https://www.virtualbox.org/

abrichr commented:
@mobley-trent thank you for putting this together. All else equal I prefer to keep as much as possible in python rather than platform-specific scripts. https://github.com/OpenAdaptAI/OpenAdapt/pull/782 has implemented it this way.

Unfortunately we can only award the bounty to one PR, and while that one is not yet merged, I think it is closer than this one (owing to the fact that it is implemented in Python, and the author is able to test it out).  Just letting you know in case there is another issue you might like to address instead of this one.

Thanks again! 🙏

mobley-trent commented:
Thanks for your patience all through @abrichr 🙏
